### ISSUE TITLE
MONITOR-1.0 FEATURE OF LIBERTY STATISTIC FOR CONNECTION POOL CONNECT WAITTIME IS MISLEADING.

### ISSUE THREAD

**Issue by fwji** [OpenLiberty/open-liberty#1846]
StatisticMeter metric used by servlet's responseTime and connectionPool's wait time have value recency issues. A more time weighted approach used by MP metrics could be a good substitution. This prototype is aimed to evaluate the possibility of using MP's Time metric to replace StatistcMeter

**Comment by fwji** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemName/com.ibm.team.workitem.WorkItem/252647

An APAR 252647 is opened against this issue.

**Comment by raymondlam** [comment]
untagging release bug so it will not be in the fix list

**Comment by fwji** [comment]
Reopen the issue to reapply the changes with Gary's performance fix. Also moving some of the data type classes from websphere package to the internal ws package.

**Comment by fwji** [comment]
This issue will be handled using MpMetrics. Reverting previous changes.

**Comment by fwji** [comment]
All previous changes have been reverted. APAR 252647 remains open as it will be addressed in 4Q release with the 2nd phase of monitor-1.0 metrics migration to mpMetrics-1.1. The new approach is to create three new total counters for connectionpool's active time, waittime and servlet's response time that will aggregate the total time since server starts, and output in mpMetrics with Prometheus format. As a result, the average time of any given windows can be calculated easily using Prometheus.

I have communicated the approach and the release schedule with the customer SVENSKA HANDELSBANKE. They agreed with the new approach and the release schedule.
